src/openassetio-core/src/utils/formatter.cpp

```diff
@@ -98,28 +104,30 @@ auto fmt::formatter<openassetio::trait::TraitSet>::format(
   std::transform(traitSet.cbegin(), traitSet.cend(), std::back_inserter(quotedTraitSet),
                  [](const std::string& trait) { return fmt::format("'{}'", trait); });
 
-  return fmt::formatter<string_view>::format(
-      fmt::format("{{{}}}", fmt::join(quotedTraitSet, ", ")), ctx);
+  return formatter<string_view>::format(fmt::format("{{{}}}", join(quotedTraitSet, ", ")), ctx);
 }
 
 auto fmt::formatter<openassetio::trait::TraitSets>::format(
     const openassetio::trait::TraitSets& traitSets, format_context& ctx) const
     -> decltype(ctx.out()) {
-  return fmt::formatter<string_view>::format(fmt::format("[{}]", fmt::join(traitSets, ", ")), ctx);
+  return formatter<string_view>::format(fmt::format("[{}]", join(traitSets, ", ")), ctx);
 }
 
 namespace {
 struct ToStringVisitor {
-  std::string operator()(openassetio::Str arg) const { return fmt::format("'{}'", arg); }
-  std::string operator()(openassetio::Float arg) const { return std::to_string(arg); }
-  std::string operator()(openassetio::Int arg) const { return std::to_string(arg); }
-  std::string operator()(openassetio::Bool arg) const { return arg ? "True" : "False"; }
+  std::string operator()(openassetio::Str arg) const;
+  std::string operator()(const openassetio::Float arg) const { return std::to_string(arg); }
+  std::string operator()(const openassetio::Int arg) const { return std::to_string(arg); }
+  std::string operator()(const openassetio::Bool arg) const { return arg ? "True" : "False"; }
 };
+std::string ToStringVisitor::operator()(openassetio::Str arg) const {
+  return fmt::format("'{}'", arg);
+}
```

Comment:
This made me stop and think... I'm not sure why the implementation got pulled outside of the struct, as it doesn't look all that different to the other overloads, with the exception of using `fmt`.

Reply:
Oh this is an interesting one, not sure about that, I might have accidentally applied the wrong automated refactor :worried: :roll_eyes:

Reply:
Yeah this was indeed an accidental auto-refactor, whoops! Reverted as part of subsequent work, force-pushed.